src/secrets/CommandEdit.ts

```diff
@@ -54,41 +54,66 @@ class CommandEdit extends CommandPolykey {
           },
           logger: this.logger.getChild(PolykeyClient.name),
         });
-        const response = await binUtils.retryAuthentication(
-          (auth) =>
-            pkClient.rpcClient.methods.vaultsSecretsGet({
-              metadata: auth,
-              nameOrId: secretPath[0],
-              secretName: secretPath[1],
-            }),
-          meta,
-        );
-        const secretContent = response.secretContent;
+        const [secretExists, secretContent] =
+          await binUtils.retryAuthentication(async (auth) => {
+            let secretExists: boolean = true;
+            let response;
+            try {
+              response = await pkClient.rpcClient.methods.vaultsSecretsGet({
+                metadata: auth,
+                nameOrId: secretPath[0],
+                secretName: secretPath[1],
+              });
+            } catch (e) {
+              if (e.cause.name === 'ErrorSecretsSecretUndefined') {
+                secretExists = false;
+              } else {
+                throw e;
+              }
+            }
+            return [secretExists, response?.secretContent];
+          }, meta);
         const tmpDir = await fs.promises.mkdtemp(
           path.join(os.tmpdir(), 'polykey-'),
         );
         const tmpFile = path.join(tmpDir, 'pksecret');
-        await this.fs.promises.writeFile(tmpFile, secretContent);
+        if (secretExists) {
+          await this.fs.promises.writeFile(tmpFile, secretContent);
+        }
         execSync(`$EDITOR \"${tmpFile}\"`, { stdio: 'inherit' });
-        let content: Buffer;
+        let content: string;
         try {
-          content = await this.fs.promises.readFile(tmpFile);
+          const tmpContent = await this.fs.promises.readFile(tmpFile);
+          content = tmpContent.toString('binary');
+          if (!secretExists) {
+            await binUtils.retryAuthentication(async (auth) => {
+              await pkClient.rpcClient.methods.vaultsSecretsNew({
+                metadata: auth,
+                nameOrId: secretPath[0],
+                secretName: secretPath[1],
+                secretContent: content,
+              });
+              await pkClient.rpcClient.methods.vaultsSecretsEdit({
+                metadata: auth,
+                nameOrId: secretPath[0],
+                secretName: secretPath[1],
+                secretContent: content,
+              });
+            });
```

Comment:
This is broken. It seems to create a new secret and then just edit it. But there is no logic branch for editing an existing secret. So editing a secret is just broken right now.

Reply:
I've fixed this in the newer commits. I will be writing test to ensure this doesn't happen again.